src/darwin/Framework/CHIP/MTRDevice_Concrete.mm

```diff
@@ -3017,7 +3053,8 @@ - (void)_setupSubscriptionWithReason:(NSString *)reason
         self.doingCASEAttemptForDeviceMayBeReachable = YES;
     }
 
-    __block bool markUnreachableAfterWait = true;
+    // Only schedule a timer for first subscriptions
+    __block bool markUnreachableAfterWait = !HadSubscriptionEstablishedOnce(self->_internalDeviceState);
 #ifdef DEBUG
     [self _callFirstDelegateSynchronouslyWithBlock:^(id testDelegate) {
         if ([testDelegate respondsToSelector:@selector(unitTestSuppressTimeBasedReachabilityChanges:)]) {
```

Comment:
This will set up a timer for non-first subscriptions too, in the test delegate case.  That seems odd.  Should this whole block only run if `markUnreachableAfterWait` is true based on the HadSubscriptionEstablishedOnce check?